js/admin.js

```diff
@@ -152,8 +152,12 @@ jQuery( document ).ready(
 		// Create the selectmenu widget only if the field is present.
 		if ( flagListExist ) {
 			// Create the jQuery UI selectmenu widget for flags list dropdown and return its instance.
-			// Need to keep the instance to be able to use it in the managment of the languages list.
-			 var selectmenuFlagList = initializeSelectmenuWidget( $( '#flag_list' ), Object.assign( {}, selectmenuOptions, selectmenuFlagListCallbacks ) );
+			var selectmenuFlagList = initializeSelectmenuWidget( $( '#flag_list' ), Object.assign( {}, selectmenuOptions, selectmenuFlagListCallbacks ) );
+			$( '#lang_list' ).on( 'notifyChange', function( event, language ) {
+				// Refresh the flag field
+				selectmenuFlagList.element.val( language.values[3] );
+				selectmenuFlagList._trigger( 'change' );
+			});
```

Comment:
Maybe we could also listen on javascript's native 'change' event, instead of triggering our custom 'notifyChange' on each `change()` callback from `selectmenuLangList`? Something like:

```javascript
$( '#lang_list' ).on(
	'change',
	function( event ) { 
		// Refresh the flag field
                var flag = parseSelectedLanguage( event ).flag;
		selectmenuFlagList.element.val( flag );
		selectmenuFlagList._trigger( 'change' );
	}
);
```

Reply:
We decide to keep a custom event and renamed it for a more pertinent one languageChanged.
Indeed the #lang_list element need to listen if the language changed